qemu/tests/virtio_serial_throttling.py

```diff
@@ -0,0 +1,64 @@
+from avocado.utils import process
+from virttest import error_context
+from virttest.utils_virtio_port import VirtioPortTest
+from qemu.tests.virtio_serial_empty_line import get_device_name_of_port, get_port_info, check_port_info
+import socket
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Induce throttling by opening chardev but not reading data from it:
+    1) Start guest with virtio-serial with unix option
+    2) On the host, open the chardev but don't read from it
+    3) In the guest, write to the virtio-serial port.
+    4) repeat step 1 to 3 with tcp option.
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+
+    def get_chardev_params(vm, serial_id):
+        serial_device = vm.devices.get(serial_id)
+        chardev_id = serial_device.get_param('chardev')
+        chardev_device = vm.devices.get(chardev_id)
+        chardev_params = chardev_device.params
+        return chardev_params
+    
+    def get_serial_params(vm, serial_id):
+        return vm.devices.get(serial_id)
+    
+    def send_data():
+        port.open()
+        port.sock.send(b'\n')
```

Comment:
It is wrong, it always send `\n`, not matter what data you create.

Reply:
but \n is still a data?

Reply:
@michiboo  Here, `port.sock.send` is: on the host, send data to the socket.  we do not need it.
What we need  for the first scenario is: send date from guest to host:
```
session.cmd_output('dd if=/dev/zero of=/dev/%s' % chardev_name)
```

Reply:
why? on polarion it said 4.repeat step 2 to step 3 when transfer data from host to guest

Reply:
@michiboo I am talking about the first scenario, not the second one. I misunderstand your purpose.
If you add this function for the testing of step4, send data from host to guest with `port.sock.send`, it is correct.
But i think `\n` is not enough.